src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,54 @@
+package nextstep.subway.application;
+
+import java.util.ArrayList;
+import java.util.List;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import org.springframework.data.crossstore.ChangeSetPersister.NotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private LineRepository lineRepository;
+
+    public LineService(LineRepository lineRepository) {
+        this.lineRepository = lineRepository;
+    }
+
+    public LineResponse readLine(Long id) throws NotFoundException {
+        Line line = lineRepository.findById(id).orElseThrow(NotFoundException::new);
+        return LineResponse.from(line);
+    }
+
+    public List<LineResponse> readLines() {
+        List<LineResponse> lineResponses = new ArrayList<>();
+        List<Line> lines = lineRepository.findAll();
+        for (Line line : lines) {
+            lineResponses.add(LineResponse.from(line));
+        }
+        return lineResponses;
```

Comment:
여기선 stream 을 사용하면 좀 더 가독성이 좋을 것 같아요

```suggestion
        return lineRepository.findAll().stream()
                .map(line -> LineResponse.from(line))
                .collect(toList());
```

Reply:
넵 수정하였습니다~